### PR TITLE
[MAINT] use tmp_path fixture in tests that do not depend on fake fmri data generation

### PR DESCRIPTION
**Pull request by Remi-Gau**
- relates to https://github.com/nilearn/nilearn/issues/3726
- extracted from [WIP] rely more on tmp_path in testing Do Not REVIEW  #3891

### CODE REVIEW

File: nilearn/datasets/tests/test_neurovault.py [hunk 1]
```diff
@@ -275,19 +274,18 @@ def test_append_filters_to_query():
...
-    with tempfile.TemporaryDirectory() as temp_dir:
-        with open(os.path.join(temp_dir, "test_nv.txt"), "w"):
-            pass
-        with pytest.raises(ValueError):
-            neurovault._get_batch(
-                f"file://{os.path.join(temp_dir, 'test_nv.txt')}",
-            )
+    with open(tmp_path / "test_nv.txt", "w"):
+        pass
+    with pytest.raises(ValueError):
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_neurovault.py:285]: extract error testing in separate tests

File: nilearn/datasets/tests/test_neurovault.py [hunk 2]
```diff
@@ -479,63 +477,57 @@ def test_result_filter_combinations():
...
+    for i, file_name in enumerate(words_files):
+        word_weights = np.zeros(n_im)
+        word_weights[i] = 1
+        words_dict = {"data": {"values": dict(zip(words, word_weights))}}
+        with open(file_name, "wb") as words_file:
+            words_file.write(json.dumps(words_dict).encode("utf-8"))
+    freq, _ = neurovault.neurosynth_words_vectorized(words_files)
+    assert freq.shape == (n_im, n_im)
+    assert (freq.sum(axis=0) == np.ones(n_im)).all()
+    with pytest.warns(UserWarning):
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_neurovault.py:510]: extract negative test in separate tests

File: nilearn/datasets/tests/test_neurovault.py [hunk 3]
```diff
@@ -625,41 +615,36 @@ def test_move_col_id():
...
+    os.makedirs(collection["absolute_path"])
+    download_params = {
+        "temp_dir": tmp_path,
+        "verbose": 3,
+        "fetch_neurosynth_words": True,
+    }
+    request_mocker.url_mapping["*"] = requests.RequestException()
+    neurovault._download_image_terms(image_info, collection, download_params)
+    download_params["allow_neurosynth_failure"] = False
+    with pytest.raises(RuntimeError):
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_neurovault.py:633]: extract negative test in separate tests

File: nilearn/glm/tests/test_check_events_file_uses_tab_separators.py [hunk 4]
```diff
@@ -86,27 +86,25 @@ def test_for_pandas_dataframe():
...
-    with InTemporaryDirectory():
-        temp_img_file = "temp_img.gif"
-        with open(temp_img_file, "wb") as temp_img_obj:
-            temp_img_obj.write(img_data)
-        with pytest.raises(ValueError):
-            _check_events_file_uses_tab_separators(events_files=temp_img_file)
+    temp_img_file = tmp_path / "temp_img.gif"
+    with open(temp_img_file, "wb") as temp_img_obj:
+        temp_img_obj.write(img_data)
+    with pytest.raises(ValueError):
```
Comment (Remi-Gau) [inline nilearn/glm/tests/test_check_events_file_uses_tab_separators.py:97]: extract negative test in separate tests

File: nilearn/glm/tests/test_check_events_file_uses_tab_separators.py [hunk 5]
```diff
@@ -86,27 +86,25 @@ def test_for_pandas_dataframe():
...
-    with InTemporaryDirectory():
-        temp_bin_file = "temp_bin.bin"
-        with open(temp_bin_file, "wb") as temp_bin_obj:
-            temp_bin_obj.write(temp_data_bytearray_from_ints)
-        with pytest.raises(ValueError):
-            _check_events_file_uses_tab_separators(events_files=temp_bin_file)
+    temp_bin_file = tmp_path / "temp_bin.bin"
+    with open(temp_bin_file, "wb") as temp_bin_obj:
+        temp_bin_obj.write(temp_data_bytearray_from_ints)
+    with pytest.raises(ValueError):
```
Comment (Remi-Gau) [inline nilearn/glm/tests/test_check_events_file_uses_tab_separators.py:106]: extract negative test in separate tests

File: nilearn/interfaces/tests/test_bids.py [hunk 6]
```diff
@@ -229,60 +228,63 @@ def test_get_bids_files_inheritance_principle_sub_folder(tmp_path, json_file):
...
+        n_sub=10,
+        n_ses=2,
+        tasks=["localizer", "main"],
+        n_runs=[1, 3],
+    )
+    # For each possible option of file selection we check that we
+    # recover the appropriate amount of files, as included in the
+    # fake bids dataset.
+
+    # 250 files in total related to subject images. Top level files like
```
Comment (Remi-Gau) [inline nilearn/interfaces/tests/test_bids.py:243]: parametrize

File: nilearn/reporting/tests/test_glm_reporter.py [hunk 7]
```diff
@@ -239,53 +239,50 @@ def _generate_img():
...
+        display_mode="z",
+        plot_type="glass",
+        table_details=table_details,
+    )
+
+
+@pytest.mark.parametrize("cut_coords", [None, (5, 4, 3)])
+def test_stat_map_to_svg_invalid_plot_type(cut_coords):
+    img = _generate_img()
+    expected_error = ValueError(
```
Comment (Remi-Gau) [inline nilearn/reporting/tests/test_glm_reporter.py:271]: use pytest raise?

File: nilearn/interfaces/tests/test_bids.py (hunk 6 above)
Comment (Remi-Gau) [reply nilearn/interfaces/tests/test_bids.py:243]: nope

**Review by ymzayek — APPROVED** [review]
LGTM thx! Just need to resolve the conflict

**Pull request merged**